src/Microsoft.ML.Api/LambdaTransform.cs

```diff
@@ -29,96 +35,42 @@ public static class LambdaTransform
-        /// is done in re-entrant fashion and without side effects.</param>
-        /// <param name="saveAction">An action that allows us to save state to the serialization stream</param>
-        /// <param name="loadFunc">A function that given the serialization stream and a data view, returns
-        /// an <see cref="ITransformTemplate"/>. The intent is, this returned object should itself be the same
-        /// as if we had recreated it using this method, but this is impossible to enforce. This transform
-        /// will do its best to save a description of this method through assembly qualified names of the defining
-        /// class, method name, and generic type parameters (if any), and then recover this same method on load,
-        /// so it should be a static non-lambda method that this assembly can legally call.</param>
-        /// <param name="inputSchemaDefinition">The optional input schema. If <c>null</c>, the schema is
-        /// inferred from the <typeparamref name="TSrc"/> type.</param>
-        /// <param name="outputSchemaDefinition">The optional output schema. If <c>null</c>, the schema is
-        /// inferred from the <typeparamref name="TDst"/> type.</param>
-        /// <typeparam name="TSrc">The type that describes what 'source' columns are consumed from the
-        /// input <see cref="IDataView"/>.</typeparam>
-        /// <typeparam name="TDst">The type that describes what new columns are added by this transform.</typeparam>
-        /// <returns>A savable mapping transform from the input source to the destination</returns>
-        public static ITransformTemplate CreateMap<TSrc, TDst>(IHostEnvironment env, IDataView source, Action<TSrc, TDst> mapAction,
-            Action<BinaryWriter> saveAction, LoadDelegate loadFunc,
-            SchemaDefinition inputSchemaDefinition = null, SchemaDefinition outputSchemaDefinition = null)
-            where TSrc : class, new()
-            where TDst : class, new()
+        // Factory for SignatureLoadModel.
+        private static ITransformer Create(IHostEnvironment env, ModelLoadContext ctx)
         {
             Contracts.CheckValue(env, nameof(env));
-            env.CheckValue(source, nameof(source));
-            env.CheckValue(mapAction, nameof(mapAction));
-            env.CheckValue(saveAction, nameof(saveAction));
-            env.CheckValue(loadFunc, nameof(loadFunc));
-            env.CheckValueOrNull(inputSchemaDefinition);
-            env.CheckValueOrNull(outputSchemaDefinition);
+            env.CheckValue(ctx, nameof(ctx));
+            ctx.CheckAtModel(GetVersionInfo());
+
+            var contractName = ctx.LoadString();
 
-            return new MapTransform<TSrc, TDst>(env, source, mapAction, saveAction, loadFunc,
-                inputSchemaDefinition, outputSchemaDefinition);
+            var composition = env.GetCompositionContainer();
+            ITransformer transformer = composition.GetExportedValue<ITransformer>(contractName);
```

Comment:
`env.GetCompositionContainer();` can return `null` (it does by default).  This will throw a NRE. We should check for a null container before trying to reference it.